src/transaction/offset.rs

```diff
@@ -195,4 +195,13 @@ impl Transaction {
                 + offset
         })
     }
+
+    /// For a serialized transaction of type `Script`, return the bytes offset
+    /// of the receipts root
+    pub const fn receipts_root_offset(&self) -> Option<usize> {
+        match self {
+            Self::Script { .. } => Some(TRANSACTION_SCRIPT_FIXED_SIZE - Bytes32::LEN),
+            Self::Create { .. } => None,
+        }
+    }
```

Comment:
Shouldn't this return only `usize` that is zero for `Create` and value for `Script`?

Reply:
`0` might be misleading since it is a valid offset, and we shouldn't have any valid receipts root offset for `Transaction::Create`. As in the PR description/commit message, this design choice was taken so the consumers of this API won't use this offset erroneously.